perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/ResourcesManagerBlImpl.java

```diff
@@ -458,24 +444,7 @@ public void assignGroupsToResource(PerunSession perunSession, Iterable<Group> gr
 
 					try {
 						getResourcesManagerImpl().assignAutomaticGroupToResource(perunSession, subgroup, resource, g);
-						GroupResourceStatus status = assignInactive ? GroupResourceStatus.INACTIVE : GroupResourceStatus.PROCESSING;
-
-						// if group is assigned with active status, we do not want to activate it again
-						if (isGroupAssigned(perunSession, resource, subgroup)) {
-							continue;
-						}
-
-						if (groupResourceAssignmentExists(perunSession, resource, subgroup)) {
-							getResourcesManagerImpl().setGroupResourceStatus(perunSession, subgroup, resource, status);
-						} else {
-							getResourcesManagerImpl().assignGroupToResourceState(perunSession, subgroup, resource, status);
-						}
-
-						// if status is not ACTIVE and should be, activate group-resource assignment
-						if (!assignInactive) {
-							// automatic subgroups-resource assignment should be async
-							activateGroupResourceAssignment(perunSession, subgroup, resource, true);
-						}
+						setAssignedGroupStatusAndActivate(perunSession, resource, true, assignInactive, subgroup);
```

Comment:
I think we can call here the new method `assignAutomaticGroupToResource`. It should be more error-resilient if we decide to change the automatic group behavior - we don't have to change it at 2 places.

Reply:
right, done